libs/langchain/langchain/smith/evaluation/runner_utils.py

```diff
@@ -1299,59 +1079,57 @@ def _evaluate_strings(self, prediction, reference=None, input=None, **kwargs) ->
     """  # noqa: E501
     if kwargs:
         warnings.warn(
-            "The following arguments are deprecated and will "
-            "be removed in a future release: "
+            "The following arguments are deprecated and "
+            "will be removed in a future release: "
             f"{kwargs.keys()}.",
             DeprecationWarning,
         )
-    wrapped_model, project_name, dataset, examples = _prepare_eval_run(
-        client, dataset_name, llm_or_chain_factory, project_name
-    )
-    results = await _arun_on_examples(
+    wrapped_model, project_name, examples, configs = _prepare_run_on_dataset(
         client,
-        examples,
-        wrapped_model,
-        concurrency_level=concurrency_level,
-        project_name=project_name,
-        verbose=verbose,
-        tags=tags,
-        evaluation=evaluation,
-        input_mapper=input_mapper,
-        data_type=dataset.data_type,
-    )
-    return TestResult(
-        project_name=project_name,
-        results=results,
+        dataset_name,
+        llm_or_chain_factory,
+        project_name,
+        evaluation,
+        tags,
+        input_mapper,
+        concurrency_level,
     )
+    progress_bar = progress.ProgressBar(len(examples))
 
+    async def _async_executor(
+        example: Example, config: RunnableConfig
+    ) -> Union[dict, str, LLMResult, ChatResult]:
+        res = await _arun_llm_or_chain(
+            example,
+            wrapped_model,
+            tags=tags,
+            callbacks=config["callbacks"],
+            input_mapper=input_mapper,
+        )
+        if verbose:
+            progress_bar.increment()
+        return res
 
-def _handle_coroutine(coro: Coroutine) -> Any:
-    """
-    Handles a coroutine from a sync context.
-
-    Args:
-        coro (asyncio.coroutine): The coroutine to be handled.
-
-    Returns:
-        any: The result of the executed coroutine.
-    """
-    # Check if there's a running event loop
-    try:
-        loop = asyncio.get_event_loop()
-    except RuntimeError:  # No event loop
-        return asyncio.run(coro)
-    if loop.is_running():
-        return loop.run_until_complete(coro)
-    else:
-        return asyncio.run(coro)
+    batch_results = await runnable_utils.gather_with_concurrency(
+        configs[0].get("max_concurrency"), *map(_async_executor, examples, configs)
+    )
+    results = _collect_test_results(examples, batch_results, configs, project_name)
+    if verbose:
+        try:
+            agg_feedback = results.get_aggregate_feedback()
+            print("\n Eval quantiles:")
```

Comment:
Why use print for some things and logger for others?

Reply:
Hm mainlyformatting? default logging configs have a prefix that makes it less pretty to view. The error messaging with logging seems right for actually reporting errors (especially since i then don't have to handle threading)

Reply:
Can switch at some point if we want